File: clean.py
import re
import sys

def clean_file(filepath, is_css=False):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    if is_css:
        content = re.sub(r'/\*.*?\*/', '', content, flags=re.DOTALL)
        content = re.sub(r'\n\s*\n', '\n\n', content)
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content.strip() + '\n')
        return

    # JS/JSX cleaning
    lines = content.split('\n')
    new_lines = []
    
    in_block_comment = False

    for line in lines:
        stripped = line.strip()
        
        # block comments logic (naively assuming no multiple block comments on one line)
        if '/*' in stripped and '*/' not in stripped and not stripped.startswith('{/*'):
            in_block_comment = True
            continue
        if in_block_comment:
            if '*/' in stripped:
                in_block_comment = False
            continue
            
        # 1. Skip lines that are entirely single-line comment
        if stripped.startswith('//') and 'eslint-disable' not in stripped:
            continue
            
        if stripped.startswith('{/*') and stripped.endswith('*/}'):
            continue
            
        # 2. Strip inline comments
        if '//' in line and not stripped.startswith('//') and 'eslint-disable' not in line:
            if 'http://' not in line and 'https://' not in line:
                parts = line.split('//')
                line = parts[0].rstrip()
                
        # 3. Strip inline JSX comments: {/* ... */}
        if '{/*' in line and '*/}' in line:
            line = re.sub(r'\{/\*.*?\*/\}', '', line)
            
        new_lines.append(line)

    # collapse multiple blank lines
    final_content = '\n'.join(new_lines)
    # final_content = re.sub(r'\n\s*\n', '\n\n', final_content)
    # wait, leaving some blank lines is fine, but more than 2 is ugly
    final_content = re.sub(r'\n{3,}', '\n\n', final_content)
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(final_content)

clean_file('src/App.jsx')
clean_file('src/index.css', is_css=True)
print("Done")

File: clean.js
import fs from 'fs';

function cleanJSX(filePath) {
    let content = fs.readFileSync(filePath, 'utf-8');
    
    const lines = content.split('\n');
    const newLines = [];
    
    let inBlockComment = false;
    
    for (let line of lines) {
        const stripped = line.trim();
        
        // Handle block comments (assuming they don't share line with code)
        if (stripped.includes('/*') && !stripped.includes('*/') && !stripped.startsWith('{/*')) {
            inBlockComment = true;
            continue;
        }
        if (inBlockComment) {
            if (stripped.includes('*/')) {
                inBlockComment = false;
            }
            continue;
        }
        
        // 1. Skip entire single-line comments
        if (stripped.startsWith('//') && !stripped.includes('eslint-disable')) {
            continue;
        }
        
        // 2. JSX block comments
        if (stripped.startsWith('{/*') && stripped.endsWith('*/}')) {
            continue;
        }
        
        // 3. Strip inline comments
        if (line.includes('//') && !stripped.startsWith('//') && !line.includes('eslint-disable')) {
            if (!line.includes('http://') && !line.includes('https://') && !line.includes('blob:')) {
                line = line.split('//')[0].trimEnd();
            }
        }
        
        // Strip inline JSX comments
        if (line.includes('{/*') && line.includes('*/}')) {
            line = line.replace(/\{\/\*.*?\*\/\}/g, '');
        }
        
        newLines.push(line);
    }
    
    // Collapse multiple blank lines
    let finalContent = newLines.join('\n');
    finalContent = finalContent.replace(/\n{3,}/g, '\n\n');
    
    fs.writeFileSync(filePath, finalContent, 'utf-8');
}

function cleanCSS(filePath) {
    let content = fs.readFileSync(filePath, 'utf-8');
    content = content.replace(/\/\*[\s\S]*?\*\//g, '');
    content = content.replace(/\n\s*\n/g, '\n\n');
    fs.writeFileSync(filePath, content.trim() + '\n', 'utf-8');
}

cleanJSX('src/App.jsx');
cleanCSS('src/index.css');

console.log('Done cleaning JS and CSS');
